### PR TITLE
Refactored methods using strings.HasPrefix

### LINKED ISSUE #92: Use strings.HasPrefix to check if strings starts with a prefix
* Inside `IssueAPIOrigin` method in `issuetracker/gitlab/gitlab.go`
* Inside `TaskURLFrom` method in `issuetracker/pivotaltracker/pivotaltracker.go`
* Inside `IssueAPIOrigin` method in `issuetracker/pivotaltracker/pivotaltracker.go`

Raised from #81 review.

### PR DESCRIPTION
**Pull request by matheusmosca**
Closes #92

### CODE REVIEW

**Comment by matheusmosca** [comment]
Hi there :wave: 

Let me know if I made some mistakes. I started learning Golang just a week ago.

File: issuetracker/pivotaltracker/pivotaltracker.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ func (it *IssueTracker) TaskModel() issuetracker.Task {
 
 // TaskURLFrom taskID returns the url for the target pivotaltracker task ID to fetch
 func (it *IssueTracker) TaskURLFrom(taskID string) string {
-	if len(taskID) > 0 && taskID[0] == '#' {
+	if len(taskID) > 0 && strings.HasPrefix(taskID, string('#')) {
```
Comment (preslavmihaylov) [inline issuetracker/pivotaltracker/pivotaltracker.go:23]: no need for the `len(taskID) > 0` check anymore. `strings.HasPrefix` [deals with that](https://play.golang.org/p/CxSXIt0YoK7).

**Review by preslavmihaylov — CHANGES_REQUESTED** [review]
Apart from the comment I left, you've missed doing this refactoring for all issue trackers.

Could you also check `issuetracker/github`? It has this issue as well.

File: issuetracker/pivotaltracker/pivotaltracker.go (hunk 1 above)
Comment (matheusmosca) [reply issuetracker/pivotaltracker/pivotaltracker.go]: Sure, this makes sense

**Comment by matheusmosca** [comment]
> Apart from the comment I left, you've missed doing this refactoring for all issue trackers.
> 
> Could you also check `issuetracker/github`? It has this issue as well.

Sure, I added the requested changes

File: issuetracker/pivotaltracker/pivotaltracker.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ func (it *IssueTracker) TaskModel() issuetracker.Task {
 
 // TaskURLFrom taskID returns the url for the target pivotaltracker task ID to fetch
 func (it *IssueTracker) TaskURLFrom(taskID string) string {
-	if len(taskID) > 0 && taskID[0] == '#' {
+	if strings.HasPrefix(taskID, string('#')) {
```
Comment (preslavmihaylov) [inline issuetracker/pivotaltracker/pivotaltracker.go:23]: Why not just write `strings.HasPrefix(taskID, "#")`?

File: issuetracker/github/github.go [hunk 3]
```diff
@@ -20,7 +20,7 @@ func (it *IssueTracker) TaskModel() issuetracker.Task {
 
 // TaskURLFrom taskID returns the url for the target github task ID to fetch
 func (it *IssueTracker) TaskURLFrom(taskID string) string {
-	if len(taskID) > 0 && taskID[0] == '#' {
+	if strings.HasPrefix(taskID, string('#')) {
```
Comment (preslavmihaylov) [inline issuetracker/github/github.go:23]: Why not just write `strings.HasPrefix(taskID, "#")`?

File: issuetracker/pivotaltracker/pivotaltracker.go [hunk 4]
```diff
@@ -30,12 +30,12 @@ func (it *IssueTracker) TaskURLFrom(taskID string) string {
 // IssueAPIOrigin returns the URL for pivotaltracker's issue-fetching API
 func (it *IssueTracker) IssueAPIOrigin() string {
 	tokens := common.RemoveEmptyTokens(strings.Split(it.Origin, "/"))
-	if tokens[0] == "pivotaltracker.com" {
+	if strings.HasPrefix(tokens[0], "pivotaltracker.com") {
```
Comment (preslavmihaylov) [inline issuetracker/pivotaltracker/pivotaltracker.go:33]: This is actually not the same as the original implementation. We want the string to be exactly `pivotaltracker.com`, rather than having the prefix.

Please revert this change

File: issuetracker/pivotaltracker/pivotaltracker.go [hunk 5]
```diff
@@ -30,12 +30,12 @@ func (it *IssueTracker) TaskURLFrom(taskID string) string {
...
 		tokens = append([]string{"https:"}, tokens...)
 	}
 
 	scheme, project := tokens[0], tokens[3]
-	if tokens[2] == "n" {
+	if strings.HasPrefix(tokens[2], "n") {
```
Comment (preslavmihaylov) [inline issuetracker/pivotaltracker/pivotaltracker.go:38]: Not the same as original implementation. Please revert

**Review by preslavmihaylov — CHANGES_REQUESTED** [review]

**Comment by matheusmosca** [comment]
I reverted the issues that I have created. Thank you for spending your time reviewing my PR.

**Comment by sattellite** [comment]
I think all implementations of `TaskURLFrom` must be refactored the same. I like add # prefix to gitlab issues and github issues. Prefix # is defacto standard.

**Review by preslavmihaylov — APPROVED** [review]

**Comment by preslavmihaylov** [comment]
@sattellite fair enough.

I've created a follow-up for that - https://github.com/preslavmihaylov/todocheck/issues/94
